framework/src/util/Any.cpp

```diff
@@ -42,26 +43,58 @@ void ThrowBadAnyCastException(const std::string& funcName,
 }
 }
 
+std::ostream& indent_line(std::ostream& os, const uint8_t increment, const int32_t indent)
+{
+  if (increment > 0) {
+    os << std::endl << std::setw(std::max(0,indent)) << ' ';
+  }
+  return os;
+}
+
 std::ostream& any_value_to_string(std::ostream& os, const Any& any)
 {
   os << any.ToString();
   return os;
 }
 
-std::ostream& any_value_to_json(std::ostream& os, const Any& val)
+std::ostream& any_value_to_json(std::ostream& os, const Any& val, const uint8_t increment, const int32_t indent)
 {
-  os << val.ToJSON();
+  os << val.ToJSON(increment, indent);
   return os;
 }
 
-std::ostream& any_value_to_json(std::ostream& os, const std::string& val)
+std::ostream& any_value_to_json(std::ostream& o, const std::string& s, const uint8_t, const int32_t)
 {
+#if NEVER
+  // The original code for this function fails to properly escape the characters in the string to
+  // make it a proper JSON string. The new code below does so.
   return os << '"' << val << '"';
+#endif
+  o << '"';
+  for (auto c = s.cbegin(); c != s.cend(); c++) {
+    switch (*c) {
+      case '"' : o << "\\\""; break;
+      case '\\': o << "\\\\"; break;
```

Comment:
cppreference says that the following are also escape sequences. \\? \\a \\v. Should we support all of them for completeness? Of those 3 that are missing the \\? is the only one that wouldn't be picked up by your default case. It's hex value is 0x3F.

Reply:
Hmm... I'm not sure. Although those characters can be escaped in c++, I don't see them as needing special treatment for JSON strings. Here's what needs to happen with them:

https://www.crockford.com/mckeeman.html

Reply:
since \?, \a, and \v are not in the JSON grammar as needing escaping we shouldn't need to handle them.